### PR TITLE
[GCP] Add GCP Metrics Data streams

### LINKED ISSUE #490: Create GCP package
# Integration release checklist

This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes

- [ ] Change follows [development guidelines](https://github.com/elastic/integrations/tree/master/doc/development/guidelines)
- [ ] Supported versions of the monitoring target are documented
- [ ] Supported operating systems are documented (if applicable)
- [ ] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [ ] Documentation exists
- [ ] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [ ] At least a manual test with ES / Kibana / Agent has been performed.
- [ ] Required Kibana version set to:

### New Package

- [ ] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes

- [ ] Dashboards exists
- [ ] Screenshots added or updated
- [ ] Datastream filters added to visualizations

### Log dataset changes

- [ ] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [ ] Generated output for at least 1 log file exists

### Metric dataset changes

_This entry is currently only recommended. It will be mandatory once we provide better support for it._

- [ ] Sample event (`sample_event.json`) exists

### Filebeat module changes

- [ ] Test log files exist for the grok patterns
- [ ] Generated output for at least 1 log file exists

### Metricbeat module changes

- [ ] Example `data.json` exists and an automated way to generate it exists (`go test -data`)
- [ ] Test environment in Docker exist for integration tests

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Migrates GCP Metricsets to GCP package

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Author's Checklist

- [ ]

## Related issues

- Closes #490

### CODE REVIEW

**Review by endorama — CHANGES_REQUESTED** [review]

File: packages/gcp/_dev/build/docs/logs.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Google Cloud Project Logs Integration
+
+The Google Cloud Project Logs integration collects and parses Google Cloud audit, VPC flow,
```
[the remark this answers is not in the document]
Comment (legoguy1000) [reply packages/gcp/_dev/build/docs/logs.md:3]: Ya, I meant Platform.

File: packages/gcp/manifest.yml [hunk 2]
```diff
@@ -23,39 +23,124 @@ screenshots:
...
-            required: false
-            show_user: true
+        # vars:
+        #   - name: alternative_host
+        #     type: text
+        #     title: Alternative host
+        #     multi: false
+        #     required: false
+        #     show_user: false
+        #     description: "GCP Alternative host"
```
Comment (legoguy1000) [inline packages/gcp/manifest.yml:79]: @endorama When I uncomment this and lines 101-124 and then try to configure the integration I get the below error.  Any ideas? I can move the alternative host setting to each data stream but I think it makes sense to keep it at this level.
image

Comment (endorama) [reply packages/gcp/manifest.yml:79]: I think this is a known missing feature, [Fleet does not support input level variables](https://github.com/elastic/kibana/issues/112272). Is this the case?

Comment (legoguy1000) [reply packages/gcp/manifest.yml:79]: I mean that lines up but if you look at the current version of the GCP package, it currently has input level variables, https://github.com/elastic/integrations/blob/master/packages/gcp/manifest.yml#L33.  I haven't actually brought up a stack with the current version of the package to see if it also errors out.

Comment (endorama) [reply packages/gcp/manifest.yml:79]: You can have 1 input with input level packages, but multiple inputs are not supported. So the solution is to move variables to the package level instead.

**Comment by endorama** [comment]
Hello @legoguy1000, I spent some time on this PR but the changes is very big and is difficult to review. Moreover I would like to add tests alongside all these changes, but this proves difficult.

As I really value this contribution and I would like to move it forward, here is my proposal: I created a new branch called [`gcp-metrics`](https://github.com/elastic/integrations/tree/gcp-metrics). May you to split this PR into multiple ones (one for each datastream/metricset) and target `gcp-metrics` as base branch?
Doing that would allow better review and being able to merge and add tests in smaller batches, which in the end will speed up inclusion.

What do you think? From your side the only additional effort would be to split this PR, as then I would be able add the missing pieces as needed.

Thank you!

File: packages/gcp/manifest.yml (hunk 2 above)
Comment (legoguy1000) [reply packages/gcp/manifest.yml:79]: ok, i figured it was something like that.

**Comment by legoguy1000** [comment]
Do you want me to make each log/metricset its own policy template so they can show up individually like AWS or like I had it and just group Logs and Metrics?

**Comment by endorama** [comment]
@legoguy1000 As in the AWS integration I expect each data stream will require it's own policy, but I'll check internally. In any case as we are targetting a dedicated branch I can help performing these changes after merge :)

**Pull request closed without merging**